### ISSUE TITLE
local_repository doesn't accept "-" in name

### ISSUE THREAD

**Issue by dmivankov** [bazelbuild/bazel#12612]
### Description of the problem / feature request:

Setting name to "a-b" fails with
> new_local_repository rule //external:a-b's name field must be a legal workspace name; workspace names may contain only A-Z, a-z, 0-9, '-', '_' and '.'

Error message is confusing, also other repository rules accept "-". Replacing "a-b" with "ab" makes repository work

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

WORKSPACE
```
new_local_repository(
  name = "a-b",
  path = "abcdef",
)
```
then
```sh
bazel query //external:a-b
```

### What operating system are you running Bazel on?

NixOS

### What's the output of `bazel info release`?

release 3.3.1- (@non-git)

### If `bazel info release` returns "development version" or "(@non-git)", tell us how you built Bazel.

NixOS package set

**Comment by dmivankov** [comment]
The issue seems to be that workspaces can't have "-" in name
```
new_local_repository(
  name = "a-b",
  path = "abcdef",
  workspace_file_content = """
workspace("whatever")
""",
)
```
also doesn't work.

**Comment by dmivankov** [comment]
"a_b" seems to work

**Comment by dmivankov** [comment]
https://github.com/bazelbuild/bazel/blob/a7e4a2b539aad8240672f19ac3c23b277f7915f8/src/main/java/com/google/devtools/build/lib/packages/WorkspaceGlobals.java#L52
https://github.com/bazelbuild/bazel/blob/50013ddb3d3010d0559a7ddbb611b9caf06f73b8/src/main/java/com/google/devtools/build/lib/cmdline/RepositoryName.java#L42
one of those may be wrong + lack of tests, some potential places to add tests
https://github.com/bazelbuild/bazel/blob/5a4b538970cda4f3d60af11f052dcc69b5b2c976/src/test/shell/bazel/local_repository_test.sh
https://github.com/bazelbuild/bazel/blob/199e70c11716cccf098f06b50c98d4f44df08b53/src/test/shell/bazel/workspace_test.sh
https://github.com/bazelbuild/bazel/blob/9993785fa0c4fa4172aa31d306f3abea76833abf/src/test/java/com/google/devtools/build/lib/packages/WorkspaceFactoryTest.java

**Comment by dmivankov** [comment]
Those two regexes are fine
```java
import java.util.regex.Pattern;

public class Test {
	private static final Pattern LEGAL_WORKSPACE_NAME = Pattern.compile("^\\p{Alpha}[-.\\w]*$");
	private static final Pattern VALID_REPO_NAME = Pattern.compile("@[\\w\\-.]*");

	public static void main(String[] args) {
		System.out.println("Workspace matches: " + LEGAL_WORKSPACE_NAME.matcher(args[0]).matches());
		System.out.println("Repo matches: " + VALID_REPO_NAME.matcher("@" + args[0]).matches());
	}
}
```

```sh
$ java Test.java a-b
Workspace matches: true
Repo matches: true
```

**Comment by dmivankov** [comment]
Aha, fixed sometime after 3.3.1 where regex is
https://github.com/bazelbuild/bazel/blob/3.3.1/src/main/java/com/google/devtools/build/lib/packages/WorkspaceGlobals.java#L53
